core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -491,6 +491,101 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
     }
   }
 
+  /**
+   * Set available resources on the driver/executor.
+   */
+  private[spark] def setResources(
+      resourceName: String,
+      resourceCount: Int,
+      resourceAddresses: Option[Seq[String]],
+      isDriver: Boolean): SparkConf = {
+    if (resourceAddresses.isDefined && resourceAddresses.get.size != resourceCount) {
+      val nodeType = if (isDriver) {
+        "driver"
+      } else {
+        "executor"
+      }
+      throw new SparkException(s"Specified $resourceCount $resourceName(s) on $nodeType, while " +
+        s"the size of device addresses is ${resourceAddresses.size}.")
+    }
+    val prefix = if (isDriver) {
+      s"$SPARK_DRIVER_RESOURCE_PREFIX.$resourceName"
+    } else {
+      s"$SPARK_EXECUTOR_RESOURCE_PREFIX.$resourceName"
+    }
+    set(s"$prefix.$SPARK_RESOURCE_COUNT", resourceCount.toString)
+    if (resourceAddresses.isDefined) {
+      set(s"$prefix.$SPARK_RESOURCE_ADDRESSES", resourceAddresses.get.mkString(","))
+    }
+    this
+  }
+
+  /**
+   * Set task resource requirement.
+   */
+  private[spark] def setResourceRequirement(
```

Comment:
`setTaskResourceRequirement`

Reply:
Similar to above,its private and if its only used for testing, wouldn't it be better as test util?